sql/src/main/java/org/apache/druid/sql/calcite/planner/Rules.java

```diff
@@ -234,7 +242,11 @@ private static Program buildHepProgram(Iterable<? extends RelOptRule> rules,
     return Programs.of(builder.build(), noDag, metadataProvider);
   }
 
-  private static List<RelOptRule> druidConventionRuleSet(final PlannerContext plannerContext)
+  @VisibleForTesting
```

Comment:
Done, as discussed the annotation doesn't add a lot of value without a static checker which enforces only test usage.

Reply:
Counterpoint: I like the annotation. Its meaning is very clear: "this method is not intended to be part of the public API of the class; it's only marked as `public` (or package-private, or whatever) so it can be unit-tested". There's no need to provide context about why it's a bad idea. It's up to the person that later on wants to use it to determine why it's a good idea.

Reply:
The thing is, that's true for all code everywhere.  Whether it's public/private/package-private, it's up to the person who is using something to determine why it's a good idea to use it.  More often than not, games with accessors like this only add friction to writing code and don't really add too much value (it's not like the `@VisibleForTesting` annotation blocks the build if it's used outside of tests or otherwise restricts type-ahead for IntelliJ).

Generally speaking, any place where this annotation needs to be added is actually indicative of a problem with code structure and encapsulation as it indicates that it's not possible to adequately test the class based on the interfaces that are supposed to be interacted with.